Error: variable-sized object may not be initialized

When trying to build tag v1.0.0 I get an error:

```
> cmake -Wno-dev -DCMAKE_INSTALL_PREFIX=/home/romanov/eic/soft/irt/irt-v1.0.0 -DCMAKE_CXX_STANDARD=17   /home/romanov/eic/soft/irt/src/v1.0.0&& cmake --build . -- -j 28&& cmake --build . --target install

/home/romanov/eic/soft/irt/src/v1.0.0/source/ChargedParticle.cc:60:17: error: variable-sized object may not be initialized
        double weights[zdim+1] = {0.0};
                       ^~~~~~
```

Compilers used (from CMake configuration)

```
-- The CXX compiler identification is Clang 10.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAVE_FLAG_STD_CXX17
-- Performing Test HAVE_FLAG_STD_CXX17 - Success
-- The C compiler identification is GNU 9.4.0
```


Full log: 

[irt_compile_log.txt](https://github.com/eic/irt/files/9871915/irt_compile_log.txt)

--- a/.github/workflows/linux-eic-shell.yml
+++ b/.github/workflows/linux-eic-shell.yml
@@ -16,6 +16,12 @@ jobs:
 
   build:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { cc: gcc,   cxx: g++     }
+          - { cc: clang, cxx: clang++ }
     steps:
       - uses: actions/checkout@v3
       - uses: cvmfs-contrib/github-action-cvmfs@v3
@@ -23,12 +29,12 @@ jobs:
         with:
           platform-release: "jug_xl:nightly"
           run: |
-            cmake -S . -B build -DDELPHES=ON -DEVALUATION=OFF -DCMAKE_INSTALL_PREFIX=install
+            CC=${{matrix.cc}} CXX=${{matrix.cxx}} cmake -S . -B build -DDELPHES=ON -DEVALUATION=OFF -DCMAKE_INSTALL_PREFIX=install
             cmake --build build
             cmake --install build
       - uses: actions/upload-artifact@v3
         with:
-          name: install
+          name: install_${{matrix.cc}}
           path: install
           retention-days: 1
           if-no-files-found: error

--- a/delphes/include/DelphesConfig.h
+++ b/delphes/include/DelphesConfig.h
@@ -124,7 +124,7 @@ class EtaRange {
 class DelphesConfig {
  public:
   DelphesConfig(const char *dname);
-  ~DelphesConfig() {};
+  virtual ~DelphesConfig() {};
 
   MassHypothesis *AddMassHypothesis(int pdg, double max_contamination_left = 1.0, 
 				    double max_contamination_right = 1.0);

--- a/delphes/source/DelphesConfig.cc
+++ b/delphes/source/DelphesConfig.cc
@@ -135,7 +135,7 @@ void DelphesConfig::DetermineThresholds( void )
 	  hypo->SetThreshold(min);
       } //for erange..mrange
 
-    printf("@T@ %d -> %7.2f\n", hypo->GetThreshold());
+    printf("@T@ %d -> %7.2f\n", ih, hypo->GetThreshold());
   } //for iq
 } // DelphesConfig::DetermineThresholds()
 

--- a/include/CherenkovRadiator.h
+++ b/include/CherenkovRadiator.h
@@ -51,10 +51,12 @@ class CherenkovRadiator: public TObject {
     return m_AlternativeMaterialName.Data();
   };
 
- private:
+ protected:
   // Run-time variables for the GEANT pass;
   const G4LogicalVolume *m_LogicalVolume;          //!
   const G4RadiatorMaterial *m_Material;            //!
+
+ private:
 
   // Refractive index calculated for some fixed reference wave length (supposedly the average 
   // one as seen on the detected photon wave length plot);

--- a/include/ParametricSurface.h
+++ b/include/ParametricSurface.h
@@ -39,7 +39,7 @@ class ParametricSurface: public TObject {
     auto pt = GetSpacePoint(u, v);
     return GetNormal(pt);
   };
-  // default, not used in all derived classes
+  // default, not overriden by all derived classes
   virtual TVector3 GetNormal( void ) const {
     return TVector3(0.0, 0.0, 0.0);
   };

--- a/include/ParametricSurface.h
+++ b/include/ParametricSurface.h
@@ -39,6 +39,10 @@ class ParametricSurface: public TObject {
     auto pt = GetSpacePoint(u, v);
     return GetNormal(pt);
   };
+  // default, not used in all derived classes
+  virtual TVector3 GetNormal( void ) const {
+    return TVector3(0.0, 0.0, 0.0);
+  };
   const TVector3 &GetCenter( void ) const { return m_Center; };
   virtual double GetDistance(const TVector3 &xx) const = 0;
 

--- a/include/ReflectionPoint.h
+++ b/include/ReflectionPoint.h
@@ -10,14 +10,14 @@
 
 class ReflectionPoint: public TObject {
  public:
- ReflectionPoint(): m_Mirror(0), m_VolumeCopy(0) {};
+ ReflectionPoint(): m_Mirror(0)/*, m_VolumeCopy(0)*/ {};
  ReflectionPoint(CherenkovMirror *mirror, unsigned vcopy, const TVector3 &position, const TVector3 &momentum):
-  m_Mirror(mirror), m_VolumeCopy(vcopy), m_Position(position), m_Momentum(momentum) {};
+  m_Mirror(mirror), /*m_VolumeCopy(vcopy),*/ m_Position(position), m_Momentum(momentum) {};
   ~ReflectionPoint() {};
 
  private:
   TRef m_Mirror;
-  unsigned m_VolumeCopy;
+  //unsigned m_VolumeCopy; // FIXME: unused
   TVector3 m_Position, m_Momentum;
 
   ClassDef(ReflectionPoint, 1);

--- a/source/ChargedParticle.cc
+++ b/source/ChargedParticle.cc
@@ -57,7 +57,8 @@ void ChargedParticle::PIDReconstruction(CherenkovPID &pid)
       {
 	bool all_converged = true;
 	IRTSolution solutions[zdim+1];
-	double weights[zdim+1] = {0.0};
+	double weights[zdim+1];
+	for(unsigned iw=0; iw<zdim+1; iw++) weights[iw] = 0.0;
 	
 	photon->m_Phi[radiator] = 0.0;
 

